components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -1081,14 +1082,45 @@ public int compare(Tag o1, Tag o2) {
 		                                APIConstants.TAGS_INFO_ROOT_LOCATION +
 		                                        "/%s/description.txt";
 		String thumbnailPathPattern = APIConstants.TAGS_INFO_ROOT_LOCATION + "/%s/thumbnail.png";
+
+		//if the tenantDomain is not specified super tenant domain is used
+		if(tenantDomain == null || tenantDomain.trim() == "" ){
+			try {
+				tenantDomain = ServiceReferenceHolder.getInstance().getRealmService().getTenantManager().getSuperTenantDomain();
+			} catch (org.wso2.carbon.user.core.UserStoreException e) {
+				handleException("Cannot get super tenant domain name",e);
+			}
+		}
+
+		//get the registry instance related to the tenant domain
+		UserRegistry govRegistry = null;
+		try {
+			ServiceReferenceHolder.getInstance().getRealmService().getTenantManager().getTenantId(tenantDomain);
+			int tenantId = ServiceReferenceHolder.getInstance().getRealmService().getTenantManager()
+			                                     .getTenantId(tenantDomain);
+			RegistryService registryService =
+					ServiceReferenceHolder.getInstance()
+					                      .getRegistryService();
+			govRegistry = registryService.getGovernanceSystemRegistry(tenantId);
+
+		} catch (UserStoreException e) {
+			handleException("Cannot get tenant id for tenant domain name:"+tenantDomain,e);
+		} catch (RegistryException e) {
+			handleException("Cannot get registry for tenant domain name:"+tenantDomain,e);
```

Comment:
This has been fixed later in https://github.com/wso2/carbon-apimgt/commit/0e6afa31d36b155af32340114212c024d5732eb9